crates/rethnet_eth/src/remote/client.rs

```diff
@@ -694,15 +692,10 @@ mod tests {
                 .await
                 .expect_err("should have failed to retrieve non-existent block number");
 
-            if let RpcClientError::JsonRpcError { error, .. } = error {
-                assert_eq!(error.code, -32602);
-                assert_eq!(
-                error.message,
-                "invalid 1st argument: block_number value was not valid block tag or block number"
-            );
-                assert!(error.data.is_none());
+            if let RpcClientError::HttpStatus(error) = error {
+                assert_eq!(error.status(), Some(StatusCode::from_u16(400).unwrap()));
             } else {
-                unreachable!("Invalid error");
+                unreachable!("Invalid error: {error}");
             }
```

Comment:
It's really too bad that the error is now being reported this way. So much less information. It makes me wonder if it's a transient thing (if it might switch back to a legit RPC response at some point). And it also makes me wonder whether Infura responds "properly" (with an actual RPC response) or with an HTTP status code like this. We should make sure our stuff is compatible with (gives reasonable error messages for) Infura too. I'm eager to get this PR merged, but maybe we need a ticket to check on that in the future?

Reply:
I think it responds both with a `400` error and a text body. We could iterate on this in the future and provide both the status + response body?

Reply:
I created an issue to address this in the future: https://github.com/NomicFoundation/rethnet/issues/100